xarray/test/test_dataarray.py

```diff
@@ -352,6 +352,43 @@ def test_broadcast_equals(self):
         self.assertFalse(a.broadcast_equals(c))
         self.assertFalse(c.broadcast_equals(a))
 
+    def test_notnull_equals(self):
```

Comment:
Rather than writing such exhaustive test cases for each xarray class, consider writing comprehensive unit tests just for `array_notnull_equiv`.